todo_app/app.py

```diff
@@ -1,44 +1,85 @@
 from flask import Flask, render_template, url_for, redirect, request
-from .data.session_items import get_items, add_item
+from flask_login.utils import login_user
+from werkzeug.datastructures import Accept
+
+from todo_app.data.user import User
 import pymongo
 from datetime import datetime
 from bson.objectid import ObjectId
+from flask_login import LoginManager, login_required, current_user
+from oauthlib.oauth2 import WebApplicationClient
 
-
-#from todo_app.flask_config import Config
+from todo_app.flask_config import Config
 
 
 import requests
+import os
+import functools
+
 
 from todo_app.ToDoCard import ToDoCard
 from todo_app.ViewModel import ViewModel
 
 import os
 
+
+def writer_required(func):
+    @functools.wraps(func)
+    def writer_check():
+        if current_user.user_id != os.getenv("USER_ID"):
+            raise ValueError('Access Denied: Does Not Have Appropiate Privilages')
+            #redirect back to page or new page with 403 error?
+        else:
+            func()
+    return writer_check
+
+
 def create_app():
     app = Flask(__name__)
-    app.config.from_object('todo_app.flask_config')
+    app.config.from_object(Config())
 
-    @app.route('/hello_world')
-    def index():
-        items = get_items()
-        return render_template('index.html', items = items)
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client = WebApplicationClient(os.getenv('CLIENT_ID'))
+        github_auth_uri =  client.prepare_request_uri("https://github.com/login/oauth/authorize", redirect_uri="http://127.0.0.1:5000/login/callback")
```

Comment:
Note that this redirect_uri being hardcoded means that this obviously won't work in production! We can either pull it from an environment variable, or we can just rely on the default redirect set up at the OAuth app level and not pass anything through here (which loses that extra level of validation, but makes our lives a shade easier!)

Reply:
So just a normal environment variable in` .env `then?

Reply:
(And `.env.test`